components/Forms/CreateResourceForm/CreateResourceForm.js

```diff
@@ -0,0 +1,149 @@
+// TODO: Make radio input component
+const free = { value: 'true', label: 'Free' };
+const paid = { value: 'false', label: 'Paid' };
+
+const costOptions = [free, paid];
+
+const validationSchema = Yup.object().shape({
+  name: Yup.string().required(validationErrorMessages.required).default(''),
+  url: Yup.string().required(validationErrorMessages.required).default(''),
+  category: Yup.string().required(validationErrorMessages.required).default(''),
+  languages: Yup.array().default(['']),
+  cost: Yup.string().default(free.value),
+  notes: Yup.string().default(''),
+});
+
+CreateResourceForm.propTypes = {
+  categories: arrayOf(shape({ name: string, id: number })).isRequired,
+  languages: arrayOf(shape({ name: string, id: number })).isRequired,
+  onSuccess: func.isRequired,
+  onFailure: func.isRequired,
+};
+
+function CreateResourceForm({ categories, languages, onSuccess, onFailure }) {
+  const initialValues = React.useMemo(() => validationSchema.cast(), []);
+
+  const categoryOptions = categories.map(({ name }) => ({
+    value: name.toLowerCase(),
+    label: name,
+  }));
+
+  const languageOptions = languages.map(({ name }) => ({
+    value: name.toLowerCase(),
+    label: name,
+  }));
+
+  const handleSubmit = async ({ cost, ...rest }, actions) => {
+    const values = { ...rest, free: cost === free.value };
+
+    try {
+      await createResource(values);
+
+      actions.setSubmitting(false);
+      actions.resetForm();
+
+      onSuccess();
+    } catch (error) {
+      actions.setSubmitting(false);
+
+      onFailure(getServerErrorMessage(error));
+    }
+  };
+
+  return (
+    <Formik
+      initialValues={initialValues}
+      onSubmit={handleSubmit}
+      validationSchema={validationSchema}
+    >
+      {({ isSubmitting }) => (
+        <Form className={styles.CreateResourceForm}>
+          <Field
+            type="text"
+            name="name"
+            label="Name*"
+            placeholder="What is this resource called?"
+            component={Input}
+            disabled={isSubmitting}
+            required
+          />
+
+          <Field
+            type="url"
+            name="url"
+            label="URL*"
+            placeholder="Where can one access this resource?"
+            component={Input}
+            disabled={isSubmitting}
+            required
+          />
+
+          <Field
+            name="category"
+            label="Category*"
+            placeholder="What category best applies to this resource?"
+            options={categoryOptions}
+            component={Select}
+            isDisabled={isSubmitting}
+            required
+          />
+
+          <Field
+            name="languages"
+            label="Programming language(s)"
+            placeholder="What language or languages are involved?"
+            options={languageOptions}
+            component={Select}
+            isDisabled={isSubmitting}
+            required
```

Comment:
`languages` aren't relevant for every resource, so it's not required. I noticed you didn't put a `*` by the label, but won't this still make it fail validation if it's empty? Or does the default of `['']` cover that case and it's never empty?

Reply:
Ah shoot. This was just a copy paste mistake. It was clearly not required in the screenshots